test/unit/record/on.test.js

```diff
@@ -0,0 +1,85 @@
+import { assert, JSData, sinon } from '../../_setup'
+const { Record } = JSData
+
+describe("Record#on('changes')", function() {
+  it('Tracking changes to a single property', function(done) {
+    const Store = new JSData.DataStore()
+    const Foo = Store.defineMapper('foo', {
+      schema: {
+        properties: {
+          id: { type: 'number' },
+          name: { type: 'string', track: true }
+        }
+      },
+    })
+    const foo = Foo.createRecord()
+    const listener = sinon.stub()
+    foo.on('change', listener)
+    foo.name = 'new foo'
+    const secondSpec = function() {
+      foo.name = 'updated foo'
+      setTimeout(function () {
+        const [record, changes] = listener.args[1]
+        assert.equal(foo, record, "on 'change' listener called with record which was modified")
+        assert.isTrue(listener.calledTwice, "on 'change' listener was called when modifying a property")
+        assert.equal(Object.keys(changes.added).length, 0)
+        assert.equal(Object.keys(changes.removed).length, 0)
+        assert.equal(changes.changed.name, 'updated foo', "Only the property changed was emitted in the changeSet")
+        done()
+      }, 5)
```

Comment:
In this case, something like this might work:

```js
setImmediate(function () {...});
```

however, the tests need to work in a bunch of different browsers, and I'm not sure how cross-browser `setImmediate` is, so `setTimeout` might be just be good enough. See https://github.com/js-data/js-data/blob/master/karma.conf.js#L3